server/src/main/java/org/elasticsearch/index/mapper/XContentDataHelper.java

```diff
@@ -596,11 +618,14 @@ static byte[] encode(XContentBuilder builder) throws IOException {
             return encoded;
         }
 
-        static void decodeAndWriteXContent(XContentBuilder b, XContentType type, BytesRef r) throws IOException {
-            try (
-                XContentParser parser = type.xContent().createParser(XContentParserConfiguration.EMPTY, r.bytes, r.offset + 1, r.length - 1)
-            ) {
-                b.copyCurrentStructure(parser);
+        static void decodeAndWriteXContent(XContentParserConfiguration parserConfig, XContentBuilder b, XContentType type, BytesRef r)
+            throws IOException {
+            try (XContentParser parser = type.xContent().createParser(parserConfig, r.bytes, r.offset + 1, r.length - 1)) {
+                if (parser.nextToken() == null) {
+                    b.startObject().endObject();
```

Comment:
This logic is very generic, i am not sure this is always correct. I _think_ that it can only happen with objects and that's the idea but i am not sure i have all scenarios in my head. Worth a comment?